### ISSUE TITLE
int ambiguity solution force creates dicts when no pre-existing dict exists, breaking previous functionality

### ISSUE THREAD

**Issue by harel** [dpath-maintainers/dpath-python#179]
The int ambiguity fix introduced in 2.1.1 changed how `new` creates elements by assuming all numeric keys are not list indexes but dict keys. The idea of the int ambiguity PR, as I understood it, was to check if an existing key exists as a dict and if so assume the numeric value is a string and not an integer (and it's container is a dict).   

However, if the target object of `new` is empty (i.e, we're creating into a blank dict), the update assumes all numeric keys are dict keys.  

Any previous implementation that operated on an empty dict (`{}`) and relied on keys like `a/b/0/c/0/d` to result as:

`{"a": {"b": [ {"c": [ {"d": "value"} ]} ] } }`

are now broken as they appear as: 

`{"a": {"b": { "0": {"c": {"0" {"d": "value"} } } } } }`

I appreciate the value in both approaches but the intuitive method is how it was before, that numeric keys are assumed list indexes. Not to mention this particular change is subtle enough to be missed in many cases and only discovered at a later time when data corruption has already happened. 

I propose that the original functionality remains the default and the new one can be triggered by either some flat, an option or a ready made creator function.

**Comment by moomoohk** [comment]
Hi Harel,
The int ambiguity issue is a tough one, every time I think the solution is down it comes back to bite me in the ass...
I like the idea of having multiple ready made creator functions for different use cases. Better than option flags in my opinion.
Let me have a proper look at this again, hopefully for the last time.

Thanks for your valuable feedback and Shabbat Shalom

**Comment by harel** [comment]
Shalom! 
Thanks. I'm happy to jump in and have a stab at it. I agree multiple ready baked creators are nicer. The issue in this particular case is that the original PR stated it will only create dicts if a dict already exists so it's just failing that test I think, where no dict exists and a list is to be created. 

Thanks for dpath by the way. I've been introducing it to many companies I work with and it's always my "look at that thing" moment.

**Comment by harel** [comment]
I've created this fix which makes both scenarios work now. I would like to fix the relevant tests to include these as well, but otherwise this could work for now. 

```
import dpath
x={}
dpath.new(x, 'a/b/0/c/0', 'hello')
{'a': {'b': [{'c': ['hello']}]}}

y={"a": {"b": {"0": {}}}}
dpath.new(y, 'a/b/0/c/0', 'hello')
{'a': {'b': {'0': {'c': ['hello']}}}}

```

**Comment by moomoohk** [comment]
Hi @harel,
Didn't notice you had already started working on #180...
Please check out #181. It should address the issues you raised here and all tests are passing:

```pycon
>>>> new({}, "a/b/0/c/1", 1)
{'a': {'b': [{'c': [None, 1]}]}}

>>>> new({}, "a/b/0/c/0/d", 1)
{'a': {'b': [{'c': [{'d': 1}]}]}}
```

**Comment by harel** [comment]
Fantastic. Thank you! I've closed #180

**Comment by moomoohk** [comment]
@harel I've merged the changes. Please let me know if there are any other issues!
